AutoMatrix/Resources/UI/AutoMatrix.ui

```diff
@@ -0,0 +1,230 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>Matrix_bis</class>
+ <widget class="qMRMLWidget" name="Matrix_bis">
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>462</width>
+    <height>725</height>
+   </rect>
+  </property>
+  <layout class="QVBoxLayout" name="verticalLayout">
+   <item>
+    <widget class="QOpenGLWidget" name="openGLWidget"/>
+   </item>
+   <item>
+    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton" native="true">
+     <property name="text" stdset="0">
+      <string>Inputs</string>
+     </property>
+     <layout class="QFormLayout" name="formLayout_2">
+      <item row="0" column="1">
+       <layout class="QVBoxLayout" name="verticalLayout_2">
+        <item>
+         <layout class="QGridLayout" name="gridLayout_2">
+          <item row="1" column="3">
+           <widget class="QPushButton" name="SearchButtonMatrix">
+            <property name="text">
+             <string>Search</string>
+            </property>
+           </widget>
+          </item>
+          <item row="1" column="0">
+           <widget class="QLabel" name="label_4">
+            <property name="text">
+             <string>Input file(s) matrix :</string>
+            </property>
+           </widget>
+          </item>
+          <item row="0" column="3">
+           <widget class="QPushButton" name="SearchButtonPatient">
+            <property name="text">
+             <string>Search</string>
+            </property>
+           </widget>
+          </item>
+          <item row="1" column="2">
+           <widget class="QComboBox" name="ComboBoxMatrix">
+            <item>
+             <property name="text">
+              <string>File</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>Folder</string>
+             </property>
+            </item>
+           </widget>
+          </item>
+          <item row="0" column="0">
+           <widget class="QLabel" name="label_3">
+            <property name="text">
+             <string>Input file(s) patient(s) :</string>
+            </property>
+           </widget>
+          </item>
+          <item row="0" column="1">
+           <widget class="QLineEdit" name="LineEditPatient"/>
```

Comment:
See my comment on `Matrix_CLI.py` and related email. It is possible to perform the operation on MRML nodes directly, rather than saving/loading nodes to file.

You would use `qMRMLNodeComboBox`. See https://github.com/DCBIA-OrthoLab/Q3DCExtension/blob/master/Q3DC/Resources/UI/Q3DC.ui#L76-L97 for example usage.

Reply:
It seems that `qMRMLNodeComboBox` is a combo box to let the user know exactly when he enter a node, am I wrong? What we want is to let the user enter their folder with some different type of files in there (.nii.gz/.vtk/.stl/.obj/.off). Can we still do that if I add it? 
I don't understand how to use it either. I added it to my .AutoMatrix.ui file and a combo box appeared but I can't click on it to have a choice to select a node.

Reply:
`qMRMLNodeComboBox` would allow the user to work on data they already have loaded into the Slicer scene. If you intend the workflow to use files and directories instead, then the node combo box is not needed.

> I can't click on it to have a choice to select a node

I forgot this context - it is important to use `setMRMLScene` so the combo box has access to the node list. See the relevant line in Q3DC: https://github.com/DCBIA-OrthoLab/Q3DCExtension/blob/master/Q3DC/Q3DC.py#L89

Again, though, since your intended workflow deals with files and folders I don't think it's necessary to implement this.